### ISSUE TITLE
[PSA] Sunsetting Blitz

### ISSUE THREAD

**Issue by JayPanoz** [FriendsOfEpub/Blitz#66]
As one of the core (and the main) maintainer of Blitz, I’ve decided to let it go after an umpteenth event in the ebook community tonight, and sunset every project in the FriendsOfEpub workspace (the Blitz framework, the companion apps, Blitz Tasks, [Blitz Checklist](http://friendsofepub.github.io/Blitz/documentation/currentState.html), etc.). 

This decision is irrevocable – and tbh, long overdue.

I still have to lay out a plan accommodating current users, because I want to make sure those repos don’t disappear overnight. But this effectively means that all active developments are stopped tonight, current issues won’t be fixed, and all projects will eventually be archived later this year. 

This will however happen over the span of a few months. So there is still an opportunity for someone to contact me and take over maintenance and benevolent dictatorship, be it for the whole Blitz project or individual repos. I promise to help make the transition as seamless as possible, to the best of my ability.

This PSA should also be first and foremost an opportunity to thank (again) people who stepped in in 2019. So I encourage everyone to go read [the state of Blitz for last year](http://friendsofepub.github.io/Blitz/documentation/state2019.html).

It’s been an honor serving so many users, and thanks for all the fish!

🙏

**Comment by JayPanoz** [comment]
# Sunsetting Roadmap

TL;DR: All active projects in the Blitz workspace will be end-of-lifed in 2020, within 6 months, starting March 1st. This means that, on September 1st, we will archive the repos to preserve them for future forks, but they will be read-only and no longer actively maintained.

## List of Repos Impacted

- [Blitz](https://github.com/FriendsOfEpub/Blitz)
- [blitzck-cli](https://github.com/FriendsOfEpub/blitzck-cli)
- [blitz-tasks](https://github.com/FriendsOfEpub/blitz-tasks)
- [blitz-checklist](https://github.com/FriendsOfEpub/blitz-checklist)
- [eBookPerfChecklist](https://github.com/FriendsOfEpub/eBookPerfChecklist)
- [eBookDesignChecklist](https://github.com/FriendsOfEpub/eBookDesignChecklist)
- [eBookTricks](https://github.com/FriendsOfEpub/eBookTricks)

## What Happened?

I’ve been feeling caught in a constant e-publishing crossfire for years – maybe others can relate –, and an umpteenth hassle in the community made it obvious it was just not worth continuing working on Blitz in such an environment.

It’s not worth going into too much detail since this sunset has been looming over Blitz for months anyway. As highlighted in [The State of Blitz in 2019](http://friendsofepub.github.io/Blitz/documentation/state2019.html), the core maintainers are no longer actively doing ebook production, and 2019 releases would have not been possible without people stepping in. So that was eventually going to happen, the turn of events only precipitated the decision.

## End of Life Explained

We won’t delete the repos, we will use [GitHub’s Archive function](https://help.github.com/en/github/creating-cloning-and-archiving-repositories/archiving-a-github-repository) to make them read-only, signal they are no longer maintained, and preserve them for current and future forks.

This also means no new issue and pull request can be opened, contributors can only star or fork the projects.

Given Blitz’ size, I’m not expecting a new benevolent dictator to take over. I’ll gladly help make the transition as seamless as possible if that ever happens, but I’ll already be surprised if some people stepped up for 1 or 2 projects. 

There is still an opportunity to do so in the next 4 months. And do not hesitate to contact me if you want to discuss the implications: caring only about your fork might be a better option, so I’ll be happy to help if you want to make sure of that.

## Timeline

Sunsetting will happen within 6 months, starting March 1st.

There are 2 major deadlines to keep in mind: June 15th and September 1st.

### March 1st

- All issues and Pull Requests are frozen, and won’t be addressed any longer
- Issues will be labeled properly (`wontfix`), and unassigned
- Maintenance will be strictly limited to security issues
- A PSA with a link pointing to this message will be added in all impacted repos’ ReadMe
- Time will be allocated for discussions with people interested in taking over one or several repos

### June 15th

- All issues and Pull Requests are closed, including new ones
- End of call for new maintainers, as it won’t be possible to plan a seamless transition after this date
- PSA page will be created on Blitz’ website

### September 1st

- Repos archived and read-only
- Websites (Blitz, checklists, etc.) updated with a disclaimer they are no longer maintained
- End of Security maintenance

**Comment by JayPanoz** [comment]
FYI I’m currently working on a revised timeline due to the exceptional (and personal) situation caused by Covid-19. 

There has been no interest in taking over, And I think it’s fair to advance these steps a little bit, and the June timeline will be moved to May 15st, while September 1st will be moved to July 1st. 

In the current context, it’s not really worth keeping the original timeline as-is. More importantly I have to take care of family so a revised timeline is pretty much the best I can do w/o help right now. 

Stay safe.

**Comment by pkra** [comment]
@JayPanoz close it down if it helps you. People can still fork it later.

All the best wishes to you and yours!

**Comment by JayPanoz** [comment]
Thanks for your concern and kind words. 

It’s been a tough day for the eprdctn community so I switched to the revised timeline (see edited recap above). Sadly I learnt a few hours ago that I won’t have a lot of time to handle things Blitz in the upcoming days/weeks.

So in practice the May 15 deadline has been already reached, and I’ve set everything up for the July 1 deadline – need that time anyway to handle stuff in a few private repos of mine. And I’ll take care of security issues since this should be relatively easy thanks to dependabot (it’s still possible to do PRs in all repos though, so if you’re willing to help with that, it will be greatly appreciated). 

On a related note, Blitz recently switched to a personal fork of `epub-zip` due to the original repo being basically unmaintained – and my PR was not merged. Although I’m using this fork in another repo with super low maintenance, it is subpar to say the least. So if anyone has time to switch to another dependency for packaging the EPUB template, a PR would be once again warmly welcome.

**Comment by JayPanoz** [comment]
👋
